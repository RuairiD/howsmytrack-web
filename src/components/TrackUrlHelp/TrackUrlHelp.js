import React from 'react';

import { Icon, List, Typography } from 'antd';

type Props = {};

class TrackUrlHelp extends React.Component<Props> {
    /*
     * Component for displaying help page with instructions on how to getthe
     * correct track URL for each platform for use on howsmytrack
     */

    platforms = [
        {
            name: "Soundcloud",
            content: (
                <Typography.Paragraph>
                    <ol>
                        <li>Navigate to your track's page (the URL should be something like `https://soundcloud.com/ruairidx/grey`</li>
                        <li>Click the <Typography.Text strong>Share</Typography.Text> button below the track waveform.</li>
                        <li>Copy the URL below <Typography.Text strong>Private Share</Typography.Text> (or just <Typography.Text strong>Share</Typography.Text> if your track is already public)</li>
                    </ol>
                    Your URL should look like:
                    <ul>
                        <li><Typography.Text code>https://soundcloud.com/ruairidx/grey</Typography.Text></li>
                        <li><Typography.Text code>https://soundcloud.com/ruairidx/grey/secret</Typography.Text></li>
                    </ul>
                </Typography.Paragraph>
            ),
        },
        {
            name: "Google Drive",
            content: (
                <Typography.Paragraph>
                    <ol>
                        <li>Right click the track file (ensure it is the track itself and not its folder etc.)</li>
                        <li>Click the <Typography.Text strong>Share</Typography.Text> button in the right click menu</li>
                        <li>
                            Click <Typography.Text strong>Copy Link</Typography.Text>
                            <ul>
                                <li>You may need to click <Typography.Text strong>Get shareable link</Typography.Text> in the top right corner before the link is available.</li>
                            </ul>
                        </li>
                    </ol>
                    Your URL should look like:
                    <ul>
                        <li><Typography.Text code>https://drive.google.com/file/d/abcdefghijklmnopqrstuvwxyz1234567/view</Typography.Text></li>
                    </ul>
                </Typography.Paragraph>
            ),
        },
        {
            name: "Dropbox",
            content: (
                <Typography.Paragraph>
                    <ol>
                        <li>Click the <Icon type="ellipsis" /> next to your track file</li>
                        <li>Click the <Typography.Text strong>Share</Typography.Text> button in the menu</li>
                        <li>
                            Click <Typography.Text strong>Copy Link</Typography.Text> at the bottom of the popup.
                            <ul>
                                <li>You may need to click <Typography.Text strong>Create link</Typography.Text> before the link is available.</li>
                            </ul>
                        </li>
                    </ol>
                    Your URL should look like:
                    <ul>
                        <li><Typography.Text code>https://www.dropbox.com/s/abcdefghijklmno/filename</Typography.Text></li>
                    </ul>
                </Typography.Paragraph>
            ),
        },
        {
            name: "OneDrive",
            content: (
                <Typography.Paragraph>
                    <ol>
                        <li>Navigate to your file's page; the track should be playable and a media player should be visible.</li>
                        <li>Click the <Typography.Text strong>Share</Typography.Text> button in the top left corner</li>
                        <li>Click the <Typography.Text strong>Copy</Typography.Text> button next to the shareable URL. <Typography.Text strong>Do not use this URL; howsmytrack.com will reject it. Keep reading.</Typography.Text></li>
                        <li>Paste the URL into an incognito/private browsing window, or any window where you aren't logged in on OneDrive (this is important). The URL will resolve into a longer URL; copy this link from your address bar.</li>
                    </ol>
                    Your URL should look like:
                    <ul>
                        <li><Typography.Text code>https://onedrive.live.com/?authkey=AUTHKEY&cid=CID&id=ID</Typography.Text></li>
                    </ul>
                </Typography.Paragraph>
            ),
        },
    ]

    render() {
        return (
            <div>
                <Typography.Paragraph>These instructions are for desktop and may differ on mobile.</Typography.Paragraph>
                <List
                    itemLayout="vertical"
                    dataSource={this.platforms}
                    renderItem={platform => (
                        <List.Item>
                            <List.Item.Meta
                                title={platform.name}
                            />
                            {platform.content}
                        </List.Item>
                    )}
                />
            </div>
        );
    }
}

export default TrackUrlHelp;