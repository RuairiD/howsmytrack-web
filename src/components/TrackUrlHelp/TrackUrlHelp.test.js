import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TrackUrlHelp from "./TrackUrlHelp";

describe("TrackUrlHelp", () => {
    it("displays a help page for finding the user's track's url", () => {
        const wrapper = shallow(
            <TrackUrlHelp />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders individual platform's instructions in the list", () => {
        const wrapper = mount(
            <TrackUrlHelp />,
        );
        expect(wrapper.find("Item").length).toBe(4);
    });
});
