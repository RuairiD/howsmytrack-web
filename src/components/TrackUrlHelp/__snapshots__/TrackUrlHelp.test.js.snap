// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackUrlHelp displays a help page for finding the user's track's url 1`] = `
<Div
  className=""
>
  <Paragraph>
    These instructions are for desktop and may differ on mobile.
  </Paragraph>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "content": <Paragraph>
            <Ol
              className=""
            >
              <Li
                className=""
              >
                Navigate to your track's page (the URL should be something like \`https://soundcloud.com/ruairidx/grey\`
              </Li>
              <Li
                className=""
              >
                Click the 
                <Text
                  strong={true}
                >
                  Share
                </Text>
                 button below the track waveform.
              </Li>
              <Li
                className=""
              >
                Copy the URL below 
                <Text
                  strong={true}
                >
                  Private Share
                </Text>
                 (or just 
                <Text
                  strong={true}
                >
                  Share
                </Text>
                 if your track is already public)
              </Li>
            </Ol>
            Your URL should look like:
            <Ul
              className=""
            >
              <Li
                className=""
              >
                <Text
                  code={true}
                >
                  https://soundcloud.com/ruairidx/grey
                </Text>
              </Li>
              <Li
                className=""
              >
                <Text
                  code={true}
                >
                  https://soundcloud.com/ruairidx/grey/secret
                </Text>
              </Li>
            </Ul>
          </Paragraph>,
          "name": "Soundcloud",
        },
        Object {
          "content": <Paragraph>
            <Ol
              className=""
            >
              <Li
                className=""
              >
                Right click the track file (ensure it is the track itself and not its folder etc.)
              </Li>
              <Li
                className=""
              >
                Click the 
                <Text
                  strong={true}
                >
                  Share
                </Text>
                 button in the right click menu
              </Li>
              <Li
                className=""
              >
                Click 
                <Text
                  strong={true}
                >
                  Copy Link
                </Text>
                <Ul
                  className=""
                >
                  <Li
                    className=""
                  >
                    You may need to click 
                    <Text
                      strong={true}
                    >
                      Get shareable link
                    </Text>
                     in the top right corner before the link is available.
                  </Li>
                </Ul>
              </Li>
            </Ol>
            Your URL should look like:
            <Ul
              className=""
            >
              <Li
                className=""
              >
                <Text
                  code={true}
                >
                  https://drive.google.com/file/d/abcdefghijklmnopqrstuvwxyz1234567/view
                </Text>
              </Li>
            </Ul>
          </Paragraph>,
          "name": "Google Drive",
        },
        Object {
          "content": <Paragraph>
            <Ol
              className=""
            >
              <Li
                className=""
              >
                Click the 
                <Icon
                  type="ellipsis"
                />
                 next to your track file
              </Li>
              <Li
                className=""
              >
                Click the 
                <Text
                  strong={true}
                >
                  Share
                </Text>
                 button in the menu
              </Li>
              <Li
                className=""
              >
                Click 
                <Text
                  strong={true}
                >
                  Copy Link
                </Text>
                 at the bottom of the popup.
                <Ul
                  className=""
                >
                  <Li
                    className=""
                  >
                    You may need to click 
                    <Text
                      strong={true}
                    >
                      Create link
                    </Text>
                     before the link is available.
                  </Li>
                </Ul>
              </Li>
            </Ol>
            Your URL should look like:
            <Ul
              className=""
            >
              <Li
                className=""
              >
                <Text
                  code={true}
                >
                  https://www.dropbox.com/s/abcdefghijklmno/filename
                </Text>
              </Li>
            </Ul>
          </Paragraph>,
          "name": "Dropbox",
        },
        Object {
          "content": <Paragraph>
            <Ol
              className=""
            >
              <Li
                className=""
              >
                Navigate to your file's page; the track should be playable and a media player should be visible.
              </Li>
              <Li
                className=""
              >
                Click the 
                <Text
                  strong={true}
                >
                  Share
                </Text>
                 button in the top left corner
              </Li>
              <Li
                className=""
              >
                Click the 
                <Text
                  strong={true}
                >
                  Copy
                </Text>
                 button next to the shareable URL. 
                <Text
                  strong={true}
                >
                  Do not use this URL; howsmytrack.com will reject it. Keep reading.
                </Text>
              </Li>
              <Li
                className=""
              >
                Paste the URL into an incognito/private browsing window, or any window where you aren't logged in on OneDrive (this is important). The URL will resolve into a longer URL; copy this link from your address bar.
              </Li>
            </Ol>
            Your URL should look like:
            <Ul
              className=""
            >
              <Li
                className=""
              >
                <Text
                  code={true}
                >
                  https://onedrive.live.com/?authkey=AUTHKEY&cid=CID&id=ID
                </Text>
              </Li>
            </Ul>
          </Paragraph>,
          "name": "OneDrive",
        },
      ]
    }
    itemLayout="vertical"
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</Div>
`;
